Assignment 4 - 6.4 (Joe's Automative)

smokethecookie opened the pull request:
Programming question implemented.

smokethecookie committed:
Assignment 4 - Progrraming question from chapter 6.4 (Joe's Automative) implemented.

pepper-classum commented:
Also, it would be great if you can add prefixes to your branches!
Something like "feature" that comes before the "/" actually groups the branches which makes the git less confusing. Not a must but it's good practice :) 
The README has more info about this as well.

yjshin229 commented:
Also, it would be great if you can add prefixes to your branches!
Something like "feature" that comes before the "/" actually groups the branches which makes the git less confusing. Not a must but it's good practice :)
The README has more info about this as well.

yjshin229 commented on hw04/6.4JoesAutomative/Form1.cs:
```diff
@@ -0,0 +1,133 @@
+        }
+
+        private decimal PartsCharges()
+        {
```
Can i ask you why you used try-catch here?

smokethecookie replied:
I use a try and catch block because anything other than a integer or decimal inputted in the textbox will raise an error. I catch these errors and return 0 instead, so the overall calculation is not affected. For example, if the input is "alim", an error would be raised because the string "alim" cannot be transformed into a decimal. It is instead replaced with 0 and the calculation is not interrupted. The same applies for your next comment in the function LaborCharges.

yjshin229 replied:
I see.
Do you think it'll be better if we show a messageBox saying that it's an invalid input in the catch phase?
I feel like as an user I would be confused if I get a 0 returned without telling me why.

smokethecookie replied:
I tried doing that actually. I can look at it again. There was an issue with the btnCalc_Click function continuing to run because we catch the error, so the messageBox appears four times as we have four textboxes in summary.

smokethecookie replied:
I think it's fine this way as everything stated in the textbook is implemented. If you really think it might cause an issue, I can look at it tomorrow (Tuesday).

yjshin229 replied:
There is also a way to only accept numbers. such as..
 private void inputA_KeyPress(object sender, KeyPressEventArgs e)
 {
     if (!char.IsControl(e.KeyChar) && !char.IsDigit(e.KeyChar))
     {
         e.Handled = true;
     }
 }

yjshin229 commented on hw04/6.4JoesAutomative/Form1.cs:
```diff
@@ -0,0 +1,133 @@
+        private decimal LaborCharges()
+        {
+            try
+            {
```
Here as well

smokethecookie replied:
Explained in the previous comment block on line 59.

smokethecookie commented:
> Also, it would be great if you can add prefixes to your branches! Something like "feature" that comes before the "/" actually groups the branches which makes the git less confusing. Not a must but it's good practice :) The README has more info about this as well.

I will do it from now on.

yjshin229 commented on hw04/6.4JoesAutomative/Form1.cs:
```diff
@@ -0,0 +1,133 @@
+
+        private void btnCalc_Click(object sender, EventArgs e)
+        {
+            tbServSum.Text = ServiceCharges().ToString("c");
```
I have my computer set locale to Korea. 
When I run the program it shows the currency for my locale. 
![image](https://github.com/yjshin229/CS392/assets/78634177/214bea3e-b767-42e4-ae07-23d72917a87b)
It might not be a big deal, but I think we could use the following just to be safe.
value.ToString("C3", CultureInfo.CreateSpecificCulture("en-US"))

smokethecookie committed:
1. Folder Youngjin created was deleted. My folder was renamed to JoesAutomative6.4.
2. Dollar symbol ($) set as default currency type. Instead of format 'c', we are using CultureInfo.CreateSpecificCulture.
3. Textbox values are validated. If not, messagebox appears and the form is cleared.
